### PR TITLE
Add API to check if team has repo access

### LINKED ISSUE #19452: [API] check if repository is added to team
### Feature Description

A simple api to check if a team has a given repo assigned would be nice, similar to the member check api.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by qwerty287**
Add an API endpoint which checks if a team has repository access. 
Closes https://github.com/go-gitea/gitea/issues/19452

### CODE REVIEW

File: routers/api/v1/org/team.go [hunk 1]
```diff
@@ -558,6 +558,55 @@ func GetTeamRepos(ctx *context.APIContext) {
...
+	//   "200":
+	//     "$ref": "#/responses/Repository"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+
+	repo := getRepositoryByParams(ctx)
+	if ctx.Written() {
+		return
+	}
+
```
Comment (Gusted) [inline routers/api/v1/org/team.go:595]: I think we want to have a permission check beforehand to ensure the `Doer` has permission to even check if the given team has access to this repo.

Comment (qwerty287) [reply routers/api/v1/org/team.go:595]: Isn't that done by the api router which requires e.g. a team membership?

Comment (Gusted) [reply routers/api/v1/org/team.go:595]: I missed that completely.

**Review by Gusted — APPROVED** [review]

File: routers/api/v1/org/team.go [hunk 2]
```diff
@@ -558,6 +558,55 @@ func GetTeamRepos(ctx *context.APIContext) {
...
+	if !organization.HasTeamRepo(ctx, ctx.Org.Team.OrgID, ctx.Org.Team.ID, repo.ID) {
+		ctx.NotFound()
+		return
+	}
+
+	access, err := models.AccessLevel(ctx.Doer, repo)
```
Comment (lunny) [inline routers/api/v1/org/team.go:601]: Looks like it's unrelated with team?

Comment (qwerty287) [reply routers/api/v1/org/team.go:601]: Yes, that's used to get the repo permissions for the specific user in the result JSON AFAIK. If it shouldn't be there, I can remove it.

**Review by delvh — APPROVED** [review]

**Review by 6543 — APPROVED** [review]
